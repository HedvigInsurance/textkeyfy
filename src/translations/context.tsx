import * as React from 'react'
import { Language } from './types'

export const TranslationsContext = React.createContext<Language>({
  textKeys: {},
})
