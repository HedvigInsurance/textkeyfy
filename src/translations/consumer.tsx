import * as React from 'react'
import { TranslationsContext } from './context'

interface ConsumerProps {
  children: (text: string) => React.ReactNode
  textKey: string
}

export const TranslationsConsumer: React.SFC<ConsumerProps> = ({
  children,
  textKey,
}) => (
  <TranslationsContext.Consumer>
    {({ textKeys }) => {
      return children(textKeys[textKey] || textKey)
    }}
  </TranslationsContext.Consumer>
)
