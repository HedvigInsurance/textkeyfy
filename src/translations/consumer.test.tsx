import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'

import { TranslationsConsumer } from './consumer'
import { TranslationsContext } from './context'

test('should render correct text key', () => {
  const testFn = jest.fn()

  const wrapper = mount(
    <TranslationsContext.Provider value={{ textKeys: { mock: 'test' } }}>
      <>
        <TranslationsConsumer textKey="mock">
          {(text) => {
            expect(text).toEqual('test')
            testFn()

            return <></>
          }}
        </TranslationsConsumer>
      </>
    </TranslationsContext.Provider>,
  )

  expect(toJson(wrapper)).toMatchSnapshot()
  expect(testFn).toHaveBeenCalled()
})
