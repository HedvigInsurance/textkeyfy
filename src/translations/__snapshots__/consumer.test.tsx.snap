// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct text key 1`] = `
<Component
  textKey="mock"
/>
`;
